lib/dry/logic/result.rb

```diff
@@ -56,6 +56,19 @@ def success?
         end
       end
 
+      class Result::LazyValue < Result
+        def to_ary
+          [:input, [rule.name, input, [rule.to_ary]]]
+        end
+        alias_method :to_a, :to_ary
+
+        def input
+          # I think that it's better to return @input instead of nil.
+          # It will demonstrate which input doesn't have appropriate key or attribute
+          success? ? rule.evaluate_input(@input) : nil
```

Comment:
No. We should return the input. Otherwise we'd loose an important piece of
information
On Thu, 14 Jan 2016 at 15:08, Alexey Zhaboyedov <notifications@github.com>
wrote:

> In lib/dry/logic/result.rb
> <https://github.com/dryrb/dry-logic/pull/1#discussion_r49729031>:
>
> > @@ -56,6 +56,19 @@ def success?
> >          end
> >        end
> >
> > +      class Result::LazyValue < Result
> > +        def to_ary
> > +          [:input, [rule.name, input, [rule.to_ary]]]
> > +        end
> > +        alias_method :to_a, :to_ary
> > +
> > +        def input
> > +          # I think that it's better to return @input instead of nil.
> > +          # It will demonstrate which input doesn't have appropriate key or attribute
> > +          success? ? rule.evaluate_input(@input) : nil
>
> @solnic <https://github.com/solnic> What do you think about the comment
> above? Should we return nil in case of failure?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dryrb/dry-logic/pull/1/files#r49729031>.
>